stdlib/public/core/FloatingPoint.swift.gyb

```diff
@@ -2374,6 +2374,113 @@ extension BinaryFloatingPoint {
   */
 }
 
+% for Range in ['Range', 'ClosedRange']:
+%   exampleRange = '10.0..<20.0' if Range == 'Range' else '10.0...20.0'
+extension BinaryFloatingPoint
+where Self.RawSignificand : FixedWidthInteger,
+      Self.RawSignificand.Stride : SignedInteger & FixedWidthInteger,
+      Self.RawSignificand.Magnitude : UnsignedInteger {
+
+  /// Returns a random value within the specified range, using the given
+  /// generator as a source for randomness.
+  ///
+  /// Use this method to generate a floating-point value within a specific
+  /// range when you are using a custom random number generator. This example
+  /// creates three new values in the range `${exampleRange}`.
+  ///
+  ///     for _ in 1...3 {
+  ///         print(Double.random(in: ${exampleRange}, using: &myGenerator))
+  ///     }
+  ///     // Prints "18.1900709259179"
+  ///     // Prints "14.2286325689993"
+  ///     // Prints "13.1485686260762"
+  ///
+  /// The `random(in:using:)` static method chooses a random value from a
+  /// continuous uniform distribution in `range`, and then converts that value
+  /// to the nearest representable value in this type. Depending on the size and
+  /// span of `range`, some concrete values may be represented more frequently
+  /// than others.
+  ///
+  /// - Parameters:
+  ///   - range: The range in which to create a random value.
+%   if Range == 'Range':
+  ///     `range` must not be empty.
+%   end
+  ///   - generator: The random number generator to use when creating the
+  ///     new random value.
+  /// - Returns: A random value within the bounds of `range`.
+  @inlinable
+  public static func random<T: RandomNumberGenerator>(
+    in range: ${Range}<Self>,
+    using generator: inout T
+  ) -> Self {
+%   if 'Closed' not in Range:
+    _precondition(
+      range.lowerBound != range.upperBound,
+      "Can't get random value with lowerBound == upperBound"
+    )
+%   end
```

Comment:
Instead of comparing bounds inside a GYB block, you might be able to use:

```swift
_precondition(!range.isEmpty, "Can't get random value with empty range")
```

With closed ranges, the [`isEmpty`](https://developer.apple.com/documentation/swift/closedrange/1779189-isempty) property is always false.